### ISSUE TITLE
Runtime error when creating a LogEntry while PAPERLESS_DISABLE_LOGIN=true

### ISSUE THREAD

**Issue by erikarvstedt** [danielquinn/paperless#334]
When `PAPERLESS_DISABLE_LOGIN=true` and a new `LogEntry` is created after a document import, for most users `User.objects.first()` in [set_log_entry](https://github.com/danielquinn/paperless/blob/68251b8be6716cf7e5d5966568946ff7e36292f6/src/documents/signals/handlers.py#L106) will be `None`.

This leads to the `LogEntry` being created with `user=None`, resulting in an error:
```
sqlite3.IntegrityError: NOT NULL constraint failed: django_admin_log.user_id
```
```
django.db.utils.IntegrityError: NOT NULL constraint failed: django_admin_log.user_id
```

Full backtrace:
```
Starting document consumer at /home/e/docs/inbox
Parsers available: RasterisedDocumentParser
Consuming /home/e/data/docs/inbox/scan.jpg
Processing sheet #1: /tmp/paperless/paperless-0_w0xbrz/convert-0000.pnm -> /tmp/paperless/paperless-0_w0xbrz/convert-0000.unpaper.pnm
OCRing the document
Parsing for eng
Detected document date 2008-02-10T00:00:00+00:00 based on string 2008-02-10
Completed
Document 20080210000000: doc8199 consumption finished
Traceback (most recent call last):
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 328, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.IntegrityError: NOT NULL constraint failed: django_admin_log.user_id
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/home/e/src/paperless/src/manage.py", line 18, in <module>
    execute_from_command_line(sys.argv)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/home/e/src/paperless/src/documents/management/commands/document_consumer.py", line 85, in handle
    self.loop(mail_delta=mail_delta)
  File "/home/e/src/paperless/src/documents/management/commands/document_consumer.py", line 105, in loop
    self.file_consumer.run()
  File "/home/e/src/paperless/src/documents/consumer.py", line 153, in run
    logging_group=self.logging_group
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/dispatch/dispatcher.py", line 193, in send
    for receiver in self._live_receivers(sender)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/dispatch/dispatcher.py", line 193, in <listcomp>
    for receiver in self._live_receivers(sender)
  File "/home/e/src/paperless/src/documents/signals/handlers.py", line 114, in set_log_entry
    object_repr=document.__str__(),
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/models/query.py", line 394, in create
    obj.save(force_insert=True, using=self.db)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/models/base.py", line 808, in save
    force_update=force_update, update_fields=update_fields)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/models/base.py", line 838, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/models/base.py", line 924, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/models/base.py", line 963, in _do_insert
    using=using, raw=raw)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/models/query.py", line 1076, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1112, in execute_sql
    cursor.execute(sql, params)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/nix/store/f6qp2a4pck9gl7v4911s8shcz84dm2qj-python3-3.6.4-env/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 328, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.IntegrityError: NOT NULL constraint failed: django_admin_log.user_id
```

**Comment by CkuT** [comment]
Hey !
I made a quickfix (I'm in the train) for this :

https://github.com/CkuT/paperless/commit/ee250484ebd6260837d5c231cee54683ab80717f

I'm open to improvements, I think we could find a better place than a migration to create the consumer user ;)

**Comment by mannp** [comment]
I am getting the same error.

@CkuT I manually added your changes and get the following;

**Traceback (most recent call last):
  File "./manage.py", line 18, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/usr/lib/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/lib/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/usr/src/paperless/src/documents/management/commands/document_consumer.py", line 85, in handle
    self.loop(mail_delta=mail_delta)
  File "/usr/src/paperless/src/documents/management/commands/document_consumer.py", line 105, in loop
    self.file_consumer.run()
  File "/usr/src/paperless/src/documents/consumer.py", line 151, in run
    logging_group=self.logging_group
  File "/usr/lib/python3.6/site-packages/django/dispatch/dispatcher.py", line 193, in send
    for receiver in self._live_receivers(sender)
  File "/usr/lib/python3.6/site-packages/django/dispatch/dispatcher.py", line 193, in <listcomp>
    for receiver in self._live_receivers(sender)
  File "/usr/src/paperless/src/documents/signals/handlers.py", line 107, in set_log_entry
    user = User.objects.get(username="consumer")
  File "/usr/lib/python3.6/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/django/db/models/query.py", line 380, in get
    self.model._meta.object_name
django.contrib.auth.models.DoesNotExist: User matching query does not exist.**

**Comment by CkuT** [comment]
It's weird, I cannot manage to reproduce this issue. Can you detail your steps ? Did you do the migration ?

**Comment by mannp** [comment]
New to the app, so edited the files you changed and rebooted....the new user was created and I tried to consume another document when it threw this error.

I have noted though that mine seems nothing to do with PAPERLESS_DISABLE_LOGIN=true as I don't don't what that is.

Perhaps the op can do the same test to see if it works for them.

**Comment by danielquinn** [comment]
The problem here is that Paperless is being installed and a user isn't being created as part of that installation.  The Docker images do this by default, but if you're running this by some other means, and you don't create a user during installation, stuff's gonna break.

So, the short term work-around is to make sure you've followed [step 5 here](http://paperless.readthedocs.io/en/latest/setup.html#standard-bare-metal).  That will make this problem go away regardless of whether or not you're using `PAPERLESS_DISABLE_LOGIN`.

@CkuT that's a neat fix: creating a user in the db with a migration that automatically is the user that did the consuming.  I'd be happy to accept it as a PR, but the format for that migration is new to me.  Is that how Django 2.x does migrations now, or is that a personal style you're using that deviates from the standard I'm used to?

**Comment by CkuT** [comment]
I found it on a stackoverflow post, but I think this is a historical solution coming from South.
A better suitable option is to use `Runpython` class (available in django 1.11 / 2.0) : https://docs.djangoproject.com/en/2.0/ref/migration-operations/#runpython

I'll try to create a PR this weekend for this.

**Comment by danielquinn** [comment]
Yeah that's the format I'm used to.  Good to know things haven't changed *too* much when I wasn't looking. :-)
